### PR TITLE
Add an Academic Year filter to course and session reports

### PR DESCRIPTION
**Pull request by jrjohnson**
This allows a user to limit the displayed results to only a specific year.

Refs #1520

WIP:
- [x] Needs translation for `'allAcademicYears': 'All Academic Years',`.

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 46e3b9b2873ce80d3796d36025b603486a606b03

https://deploy-preview-3411--ilios-frontend.netlify.com

**Comment by jrjohnson** [comment]
Click tested by @saschaben and approved. Just needs code review.

File: app/services/reporting.js [hunk 1]
```diff
@@ -127,13 +127,13 @@ export default Service.extend({
       return rhett;
     });
 
-    return resolve(mappedResults);
+    return mappedResults.filter(obj => isEmpty(year) || parseInt(obj.course.get('year')) === parseInt(year));
```
Comment (stopfstedt) [inline app/services/reporting.js]: please specify a radix of 10 as second argument when invoking `parseInt`. 
side-note: we should configure the eslinter to check for that (if that's an option).

File: app/services/reporting.js [hunk 2]
```diff
@@ -144,7 +144,7 @@ export default Service.extend({
       return rhett;
     });
 
-    return mappedResults;
+    return mappedResults.filter(obj => isEmpty(year) || parseInt(obj.course.get('year')) === parseInt(year));
```
Comment (stopfstedt) [inline app/services/reporting.js]: radix 10 please.

**Pull request merged**